### PR TITLE
[kubernetes] Add last_terminated_reason_timestamp metric

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## Proposed commit message

Please explain:

- WHAT: add last_terminated_reason_timestamp for when last_terminated_reason event happens in K8s Integration
- WHY: https://github.com/elastic/elastic-agent/issues/3802

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

1. create local k8s cluster: `kind create cluster`
2. install stack: `elastic-package-0.98.2 stack up -d -v --version=8.14.0-SNAPSHOT`
3. edit dev-tools/kubernetes/Tiltfile to run in `mode="run"` and run:
```
cd dev-tools/kubernetes
tilt up
```
4. Create a pod with a low resources.limits.memory to cause OOMKilled

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/3802

## Screenshots
Screenshot 2024-04-26 at 14 30 47

### CODE REVIEW

**Comment by gizas** [comment]
Tania minor notes:
- Add some steps on how to test this PR. 
- Add a screenshot to prove that field is present
- Add a link for the relevant elastic-agent story where this field needs also to be added

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -275,6 +275,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Add new memory/cgroup metrics to Kibana module {pull}37232[37232]
 - Add SSL support to mysql module {pull}37997[37997]
 - Add SSL support for aerospike module {pull}38126[38126]
+- Add last_terminated_reason_timestamp metric in kubernetes module {issue}3802[3802]
```
Comment (gizas) [inline CHANGELOG.next.asciidoc:278]:
```suggestion
- Add last_terminated_reason_timestamp metric in kubernetes module {pull}39200[39200] {issue}3802[3802]
```

File: metricbeat/module/kubernetes/state_container/state_container.go [hunk 2]
```diff
@@ -59,14 +59,15 @@ var (
...
-			"kube_pod_container_status_last_terminated_reason": p.LabelMetric("status.last_terminated_reason", "reason"),
+			"kube_pod_container_status_ready":                     p.BooleanMetric("status.ready"),
+			"kube_pod_container_status_restarts_total":            p.Metric("status.restarts"),
+			"kube_pod_container_status_running":                   p.KeywordMetric("status.phase", "running"),
+			"kube_pod_container_status_terminated":                p.KeywordMetric("status.phase", "terminated"),
+			"kube_pod_container_status_waiting":                   p.KeywordMetric("status.phase", "waiting"),
+			"kube_pod_container_status_terminated_reason":         p.LabelMetric("status.reason", "reason"),
+			"kube_pod_container_status_waiting_reason":            p.LabelMetric("status.reason", "reason"),
+			"kube_pod_container_status_last_terminated_reason":    p.LabelMetric("status.last_terminated_reason", "reason"),
+			"kube_pod_container_status_last_terminated_timestamp": p.Metric("status.last_terminated_reason.sec"),
```
Comment (MichaelKatsoulis) [inline metricbeat/module/kubernetes/state_container/state_container.go:70]: Here you map `kube_pod_container_status_last_terminated_timestamp`  ksm metric to elasticsearch field `kubernetes.container.status.last_terminated_reason.sec`.  But then , the field you declare in the fields.yaml is `last_terminated_reason_timestamp`. Am I missing something?

Comment (tetianakravchenko) [reply metricbeat/module/kubernetes/state_container/state_container.go]: my mistake, reverted this - https://github.com/elastic/beats/pull/39200/commits/180afdf5df6fe200849b8575a3ec9ca5af541a91

It is a leftover, I was considering other naming, not really sure regarding the name `last_terminated_reason_timestamp`. For example in state_cronjob - https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_cronjob/state_cronjob.go#L33-L34 is used `*.sec` for time metrics, but then I figured out that there is not really a pattern for time metrics, for example:
[job](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_job/state_job.go#L39-L40) - `time.*`  (type of the field: data)
[statefulset](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_statefulset/state_statefulset.go#L30) - no `time` or `sec` in name  (type: long)
[storageclass](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_storageclass/state_storageclass.go#L37) - no `time` or `sec` in name (type: date)
[namespace](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_namespace/state_namespace.go#L30) - `created.sec`, type: double

do you have any thoughts on naming?

Comment (tetianakravchenko) [reply metricbeat/module/kubernetes/state_container/state_container.go]: done in https://github.com/elastic/beats/pull/39200/commits/90d82b61952bd251c1436d0d362b8e08cac62697

**Review by MichaelKatsoulis — APPROVED** [review]

**Comment by gizas** [comment]
> @gizas I've added steps to test and the screenshot
Nice!

> regarding elastic-agent story - do you mean to add new field to the k8s integration? I've listed it 
Ok

> can you provide the use case? we already provide the last_terminated_reason and if this metric is present it already implies that some issue happened

I was thinking that the code is a number and probably for those who use APIs is better to check for specific code and if is present then to return  it. But not so important for now as not specifically asked. Skip it

**Pull request merged**